### DISCUSSION TITLE
Self-hosting Restore a CouchDB /opt/couchdb/data Backup on a new instance

### DISCUSSION THREAD

**Discussion by superbolt-x** [Budibase/budibase#4436] [Help]
Hi!

Am hosting BB with docker-compose on a AWS EC2 instance and I am trying to mount a backup of that /opt/couchdb/data couchdb volume from my BB to another AWS EC2 instance to simulate a disaster recovery.

I did this on my docker-compose: 

couchdb-service:
    restart: always
    image: ibmcom/couchdb3
    environment:
      - COUCHDB_PASSWORD=${COUCH_DB_PASSWORD}
      - COUCHDB_USER=${COUCH_DB_USER}
    ports:
      - "${COUCH_DB_PORT}:5984"
    volumes:
      - /home/ec2-user/couchdb-data/:/opt/couchdb/data:z

I basically edited the volume part to load my backup (- /home/ec2-user/couchdb-backup/:/opt/couchdb/data:z)

I had problems starting it again because of some linux permissions error to access the CouchDB files on the container - so I tried to `chmod -R 777  /home/ec2-user/couchdb-backup/`, removed my containers and redid docker-compose up to start and worked!

But it's not free of bugs - when accessing my apps I can't see the design view [am running the latest release BB 1.0.57]

Also when trying to backup the new container CouchDB the folders are keeping that permission configuration of course (so the new /opt/couchdb/data backup up is in mode 777) - and not sure this is the correct way of doing it

Can you suggest a better way of restoring from a backup?

Thanks!

**Comment by melohagan** [comment]
Hey @superbolt-x 

There is a better way to backup now through the Budibase CLI! https://docs.budibase.com/docs/backups
